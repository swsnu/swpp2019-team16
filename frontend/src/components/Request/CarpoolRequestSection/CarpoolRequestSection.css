.requestBlock{
	text-align: left;
}

.title{
	margin-bottom: 10px;
}

.from{
	border-style: solid;
	padding-left: 10px;
	margin-bottom: 10px;
}

.to{
	border-style: solid;
	padding-left: 10px;
	margin-bottom: 10px;
}

.minimumPassenger{
	border-style: solid;
	padding-left: 10px;
	margin-bottom: 10px;
}


.button{
	text-align: center;
}
