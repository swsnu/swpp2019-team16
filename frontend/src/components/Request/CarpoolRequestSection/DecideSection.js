import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import Button from '../../common/Button';

const useStyles = makeStyles({
  root: {},
});

DecideSection.propTypes = {
  onClick: PropTypes.func.isRequired,
};

function DecideSection({ onClick }) {
  const styles = useStyles();
  return (
    <div className={styles.root}>
      <Button
        id="request-submit-button"
        variant="contained"
        color="primary"
        onClick={onClick}
        children="Carpool Request"
      />
    </div>
  );
}

export default DecideSection;