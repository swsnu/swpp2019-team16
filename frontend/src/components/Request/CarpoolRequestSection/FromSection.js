import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import Checkbox from '../../common/Checkbox';
import { Typography } from '@material-ui/core';
import Map from '../../common/Map';
import MapPin from '../../common/MapPin';
import { LocationPropTypes } from '../../../types/location';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center',
  },
  title: {
    marginBottom: '30px',
  },
  content: {
    marginBottom: '25px',
  },
  map: {
    width: '100%',
    flex: 1,
    marginBottom: '25px',
  },
  noLocationSelected: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '400px',
  },
});

FromSection.propTypes = {
  fromList: PropTypes.arrayOf(LocationPropTypes).isRequired,
  selectedFrom: LocationPropTypes,
  onClickFrom: PropTypes.func.isRequired,
};

function FromSection({ fromList, selectedFrom, onClickFrom }) {
  const styles = useStyles();
  return (
    <div className={styles.root}>
      <div className={styles.title}>
        <Typography variant={'h4'}>Where are you?</Typography>
      </div>
      <div className={styles.content}>
        {fromList &&
          fromList.map(location => (
            <Checkbox
              key={location.name}
              name={'from'}
              value={location.name}
              onClick={() =>
                selectedFrom === location.name
                  ? onClickFrom(null)
                  : onClickFrom(location)
              }
              checked={
                selectedFrom !== null && selectedFrom.name === location.name
              }
            />
          ))}
      </div>
      <div className={styles.map}>
        {selectedFrom === null ? (
          <div className={styles.noLocationSelected}>
            <Typography variant={'h5'}>No location selected.</Typography>
          </div>
        ) : (
          <div>
            <Map
              center={{
                lat: selectedFrom.coordinate.lat,
                lng: selectedFrom.coordinate.lng,
              }}
              width={'100%'}
              height={'400px'}
            >
              <MapPin
                lat={selectedFrom.coordinate.lat}
                lng={selectedFrom.coordinate.lng}
              />
            </Map>
          </div>
        )}
      </div>
    </div>
  );
}

export default FromSection;
