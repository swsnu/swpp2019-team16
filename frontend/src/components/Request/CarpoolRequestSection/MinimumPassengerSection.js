import React from 'react';
import { makeStyles } from '@material-ui/styles';
import Checkbox from '../../common/Checkbox';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center',
  },
  title: {
    marginBottom: '30px',
  },
  content: {
    marginBottom: '25px',
  },
  map: {
    width: '100%',
    flex: 1,
    marginBottom: '25px',
  },
  noLocationSelected: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '400px',
  },
});

MinimumPassengerSection.propTypes = {
  minimumPassengerOptions: PropTypes.arrayOf(PropTypes.string).isRequired,
  selectedMinimumPassenger: PropTypes.string,
  onClickMinimumPassenger: PropTypes.func.isRequired,
};

function MinimumPassengerSection({
  minimumPassengerOptions,
  selectedMinimumPassenger,
  onClickMinimumPassenger,
}) {
  const styles = useStyles();
  return (
    <div className={styles.root}>
      <div className={styles.title}>
        <Typography variant={'h4'}>How many co-passengers at least?</Typography>
      </div>
      <div className={styles.content}>
        {minimumPassengerOptions &&
          minimumPassengerOptions.map(minimumPassenger => (
            <Checkbox
              key={minimumPassenger}
              name={'minimumPassenger'}
              value={minimumPassenger}
              onClick={() =>
                selectedMinimumPassenger === minimumPassenger
                  ? onClickMinimumPassenger(null)
                  : onClickMinimumPassenger(minimumPassenger)
              }
              checked={minimumPassenger === selectedMinimumPassenger}
            />
          ))}
      </div>
    </div>
  );
}

export default MinimumPassengerSection;
