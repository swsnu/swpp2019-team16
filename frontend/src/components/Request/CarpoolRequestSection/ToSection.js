import React from 'react';
import { makeStyles } from '@material-ui/styles';
import Checkbox from '../../common/Checkbox';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';
import Map from '../../common/Map';
import MapPin from '../../common/MapPin';
import { LocationPropTypes } from '../../../types/location';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center',
  },
  title: {
    marginBottom: '30px',
  },
  content: {
    marginBottom: '25px',
  },
  map: {
    width: '100%',
    flex: 1,
    marginBottom: '25px',
  },
  noLocationSelected: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '400px',
  },
});

ToSection.propsTypes = {
  toList: PropTypes.arrayOf(LocationPropTypes).isRequired,
  selectedTo: LocationPropTypes,
  onClickTo: PropTypes.func.isRequired,
};

function ToSection({ toList, selectedTo, onClickTo }) {
  const styles = useStyles();
  return (
    <div className={styles.root}>
      <div className={styles.title}>
        <Typography variant={'h4'}>Where do you want to go?</Typography>
      </div>
      <div className={styles.content}>
        {toList &&
          toList.map(location => (
            <Checkbox
              color={'secondary'}
              key={location.name}
              name={'from'}
              value={location.name}
              onClick={() =>
                selectedTo === location.name
                  ? onClickTo(null)
                  : onClickTo(location)
              }
              checked={selectedTo !== null && selectedTo.name === location.name}
            />
          ))}
      </div>
      <div className={styles.map}>
        {selectedTo === null ? (
          <div className={styles.noLocationSelected}>
            <Typography variant={'h5'}>No location selected.</Typography>
          </div>
        ) : (
          <div>
            <Map
              center={{
                lat: selectedTo.coordinate.lat,
                lng: selectedTo.coordinate.lng,
              }}
              width={'100%'}
              height={'400px'}
            >
              <MapPin
                lat={selectedTo.coordinate.lat}
                lng={selectedTo.coordinate.lng}
              />
            </Map>
          </div>
        )}
      </div>
    </div>
  );
}

export default ToSection;
