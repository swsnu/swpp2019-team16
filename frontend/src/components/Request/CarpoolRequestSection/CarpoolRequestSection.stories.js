import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { storiesOf } from '@storybook/react';

import CarpoolRequestSection from './CarpoolRequestSection';
import { ThemeProvider } from '@material-ui/styles';
import theme from '../../../lib/styles/theme';
import { CssBaseline } from '@material-ui/core';

storiesOf('Request/CarpoolRequestSection', module)
  .addDecorator(story => (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        {story()}
      </ThemeProvider>
    </BrowserRouter>
  ))
  .add('default', () => (
    <CarpoolRequestSection
      user={{}}
      fromList={['SNU station', 'Nakseongdae station', 'Nokdu Street']}
      toList={['301 building', 'Student Center', 'Dormitory three-way']}
      minimumPassenger={['2', '3', '4']}
      onClickRequest={() => {}}
    />
  ));