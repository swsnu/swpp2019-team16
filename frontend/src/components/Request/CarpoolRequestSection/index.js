import CarpoolRequestSection from './CarpoolRequestSection';
export default CarpoolRequestSection;
