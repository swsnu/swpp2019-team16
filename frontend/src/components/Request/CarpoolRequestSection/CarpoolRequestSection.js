import React, { useState } from 'react';
import clns from 'classnames';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import FromSection from './FromSection';
import ToSection from './ToSection';
import MinimumPassengerSection from './MinimumPassengerSection';
import DecideSection from './DecideSection';
import FullPage from '@fullpage/react-fullpage';
import theme from '../../../lib/styles/theme';
import { UserPropsTypes } from '../../../types/user';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
  },
  container: {
    marginTop: '25px',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  fromSection: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    paddingTop: '0',
    paddingBottom: '0',
    paddingLeft: '32px',
    paddingRight: '32px',

    backgroundColor: theme.palette.common.white,
    color: theme.palette.text.primary,
  },
  toSection: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',

    backgroundColor: theme.palette.primary.main,
    color: theme.palette.primary.contrastText,
  },
  minimumPassenger: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',

    backgroundColor: theme.palette.secondary.main,
    color: theme.palette.primary.contrastText,
  },
  decide: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',

    backgroundColor: theme.palette.common.black,
    color: theme.palette.primary.contrastText,
  },
});

CarpoolRequestSection.propTypes = {
  user: UserPropsTypes,
  fromLocationList: PropTypes.array.isRequired,
  toLocationList: PropTypes.array.isRequired,
  minimumPassengerList: PropTypes.array.isRequired,
  onCarpoolRequest: PropTypes.func.isRequired,
};

function CarpoolRequestSection({
  user,
  fromLocationList,
  toLocationList,
  minimumPassengerList,
  onCarpoolRequest,
}) {
  const styles = useStyles();
  const [fromLocation, setFromLocation] = useState(null);
  const [toLocation, setToLocation] = useState(null);
  const [minimumPassenger, setMinimumPassenger] = useState(null);

  const onSendRequest = () => {
    if (!fromLocation || !toLocation || !minimumPassenger) {
      window.alert('Please check your request');
      return;
    }

    onCarpoolRequest({
      riderId: user.id,
      from: fromLocation.name,
      to: toLocation.name,
      minimumPassenger: minimumPassenger,
    });
  };

  return (
    <FullPage
      scrollingSpeed={1000}
      render={() => {
        return (
          <FullPage.Wrapper>
            <div className={'section'}>
              <div className={styles.container}>
                <FromSection
                  fromList={fromLocationList}
                  selectedFrom={fromLocation}
                  onClickFrom={setFromLocation}
                />
              </div>
            </div>
            <div className={clns('section', styles.toSection)}>
              <div className={styles.container}>
                <ToSection
                  toList={toLocationList}
                  selectedTo={toLocation}
                  onClickTo={setToLocation}
                />
              </div>
            </div>
            <div
              className={clns('section', styles.minimumPassenger)}
              data-testid="minimumPassenger"
            >
              <MinimumPassengerSection
                minimumPassengerOptions={minimumPassengerList}
                selectedMinimumPassenger={minimumPassenger}
                onClickMinimumPassenger={setMinimumPassenger}
              />
            </div>
            <div className={clns('section', styles.decide)}>
              <DecideSection onClick={onSendRequest} />
            </div>
          </FullPage.Wrapper>
        );
      }}
    />
  );
}

export default CarpoolRequestSection;
