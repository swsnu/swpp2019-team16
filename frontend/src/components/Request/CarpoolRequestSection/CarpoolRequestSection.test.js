import React from 'react';
import { render } from '@testing-library/react';
import CarpoolRequestSection from './CarpoolRequestSection';
import { fromLocationList, toLocationList } from '../../../types/location';
import { mockRider } from '../../../types/__mock__/user';

jest.mock('@fullpage/react-fullpage', () => ({ children }) => (
  <div>{children}</div>
));

describe('<CarpoolRequestSection />', () => {
  it('SHOULD match with snapshot WHEN all data in', () => {
    const { container } = render(
      <CarpoolRequestSection
        user={mockRider}
        fromLocationList={fromLocationList}
        toLocationList={toLocationList}
        minimumPassengerList={['2', '3', '4']}
        onCarpoolRequest={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
