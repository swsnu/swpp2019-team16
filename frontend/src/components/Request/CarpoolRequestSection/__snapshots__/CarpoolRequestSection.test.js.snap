// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Request /> SHOULD match with snapshot WHEN all data in 1`] = `
<div>
  <div
    class="sc-bdVaJa bDWFJH requestBlock"
  >
    <div
      class="title"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2"
        title="Carpool Request!"
      >
        Carpool Request!
      </h2>
    </div>
    <div
      class="from"
      data-testid="from"
    >
      <h2>
        From
      </h2>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="a"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-43 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-46"
                data-indeterminate="false"
                name="from"
                type="checkbox"
                value="a"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            a
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="b"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-43 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-46"
                data-indeterminate="false"
                name="from"
                type="checkbox"
                value="b"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            b
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="c"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-43 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-46"
                data-indeterminate="false"
                name="from"
                type="checkbox"
                value="c"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            c
          </span>
        </label>
      </div>
    </div>
    <div
      class="to"
      data-testid="to"
    >
      <h2>
        To
      </h2>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="x"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-43 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-46"
                data-indeterminate="false"
                name="to"
                type="checkbox"
                value="x"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            x
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="y"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-43 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-46"
                data-indeterminate="false"
                name="to"
                type="checkbox"
                value="y"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            y
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="c"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-43 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-46"
                data-indeterminate="false"
                name="to"
                type="checkbox"
                value="c"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            c
          </span>
        </label>
      </div>
    </div>
    <div
      class="minimumPassenger"
      data-testid="minimumPassenger"
    >
      <h2>
        Minimum Passenger
      </h2>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="2"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-43 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-46"
                data-indeterminate="false"
                name="minimumPassenger"
                type="checkbox"
                value="2"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            2
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="3"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-43 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-46"
                data-indeterminate="false"
                name="minimumPassenger"
                type="checkbox"
                value="3"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            3
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="4"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-43 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-46"
                data-indeterminate="false"
                name="minimumPassenger"
                type="checkbox"
                value="4"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            4
          </span>
        </label>
      </div>
    </div>
    <div
      class="button"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Carpool Request
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
